frontend/packages/dev-console/src/components/pipelines/pipeline-overview/PipelineStartButton.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { Button } from '@patternfly/react-core';
+import { useAccessReview } from '@console/internal/components/utils';
+import { AccessReviewResourceAttributes } from '@console/internal/module/k8s';
+import { PipelineRunModel } from '../../../models';
+import { Pipeline } from '../../../utils/pipeline-augment';
+import { startPipelineModal } from '../modals';
+
+type TriggerLastRunButtonProps = {
+  pipeline: Pipeline;
+  namespace: string;
+  impersonate?;
+};
+
+const PipelineStartButton: React.FC<TriggerLastRunButtonProps> = ({
+  pipeline,
+  namespace,
+  impersonate,
+}) => {
+  const openPipelineModal = () =>
+    startPipelineModal({
+      pipeline,
+      modalClassName: 'modal-lg',
+    });
+  const defaultAccessReview: AccessReviewResourceAttributes = {
+    group: PipelineRunModel.apiGroup,
+    resource: PipelineRunModel.plural,
+    namespace,
+    verb: 'create',
+  };
+  const isAllowed = useAccessReview(defaultAccessReview, impersonate);
+
+  return (
+    isAllowed && (
+      <Button variant="secondary" onClick={openPipelineModal}>
+        Start
+      </Button>
+    )
+  );
+};
+
+export default PipelineStartButton;
```

Comment:
You need to connect this component to redux and get the `impersonate` state from it. Right now the value of impersonate will be always undefined.

Reply:
Updated.